src/service-worker.test.js

```diff
@@ -276,12 +278,63 @@ test('SW should open deep link in click handler if one is provided', () => {
   }
   clickListener(clickEvent);
 
-  // Then the deep link should be opened in a new tab
-  expect(openedWindows).toHaveLength(1);
-  expect(openedWindows[0]).toEqual('https://pusher.com');
+  return clickEvent.getWaitUntilPromise().then(() => {
+    // Then the deep link should be opened in a new tab
+    expect(clients).toContainEqual({ url: 'https://pusher.com' });
 
-  // And the notification should be closed
-  expect(clickEvent._isOpen()).toEqual(false);
+    // And the notification should be closed
+    expect(clickEvent._isOpen()).toEqual(false);
+  });
+});
```

Comment:
Because the new solution uses `Promise` chaining, the tests were running the assertions (`expect`) before the `Promise` completed.

Previously I used `await new Promise(process.nextTick)` before the assertions, which in Jest will wait until all Promises are completed, but I changed the approach to implement `getWaitUntilPromise()` for the `clickEvent`s to be more consistent with the existing tests using `FakePushEvent`:
https://github.com/pusher/push-notifications-web/blob/0beff7a2910d16a1d03f864c62408b644aa78ba7/src/service-worker.test.js#L668-L682